deepspeed/runtime/engine.py

```diff
@@ -1433,14 +1433,16 @@ def _configure_fp16_optimizer(self, optimizer):
         initial_dynamic_scale = self.initial_dynamic_scale()
         dynamic_loss_args = self.dynamic_loss_scale_args()
         clip_grad = self.gradient_clipping()
+        model_dtype, grad_accum_dtype = self.get_data_types()
         if APEX_INSTALLED:
             fused_opts = (apex.optimizers.FusedAdam, FusedAdam)
         else:
             fused_opts = FusedAdam
         if isinstance(optimizer, fused_opts) \
                 or self.optimizer_name() in [ONEBIT_ADAM_OPTIMIZER, ZERO_ONE_ADAM_OPTIMIZER]:
             if self.dynamic_loss_scale():
-                log_dist("Creating fp16 optimizer with dynamic loss scale", ranks=[0])
+                log_dist(f'Creating {model_dtype} optimizer with dynamic loss scale',
```

Comment:
For this message, we initialize only the `FP16_Optimizer` class afterwards, it is perhaps safer not to change the message in this case, no?

Reply:
Thank you! Everything looks good for me